detection_rules/rule.py

```diff
@@ -515,6 +572,60 @@ def name(self) -> str:
     def type(self) -> str:
         return self.data.type
 
+    def _post_dict_transform(self, obj: dict) -> dict:
+        """Transform the converted API in place before sending to Kibana."""
+        super()._post_dict_transform(obj)
+
+        self.add_related_integrations(obj)
+        self.add_required_fields(obj)
+        self.add_setup(obj)
+
+        # validate new field against the schema
+        rule_type = obj['type']
+        subclass = self.get_data_subclass(rule_type)
+        subclass.from_dict(obj)
+
+        return obj
+
+    def add_related_integrations(self, obj: dict) -> None:
+        """Add restricted field related_integrations to the obj"""
+        # field_name = "related_integrations"
+        ...
+
+    def add_required_fields(self, obj: dict) -> None:
+        """Add restricted field required_fields to the obj, derived from the query AST."""
+        if isinstance(self.data, QueryRuleData) and self.data.language != 'lucene':
+            index = obj.get('index') or []
+            required_fields = self.data.get_required_fields(index)
+        else:
+            required_fields = []
+
+        field_name = "required_fields"
+        if self.check_restricted_field_version(field_name=field_name):
+            sorted_required_fields = sorted(required_fields, key=lambda f: f['name'])
```

Comment:
in order to make the output of required fields always ordered, should you instead have `get_required_fields` return an `OrderedDict`

Reply:
the obj is a list of dicts. either way, I don't think its necessary.

Reply:
I made the suggestion above. If we don't do it there, we will always have to do this from the caller (for consistency)

Reply:
Instead sorted the required_fields in the `get_required_fields` method